core/app/models/spree/gallery/variant_gallery.rb

```diff
@@ -0,0 +1,38 @@
+module Spree
+  module Gallery
+    class VariantGallery
+      def initialize(variant)
+        @variant = variant
+      end
+
+      # A list of all images associated with this gallery
+      #
+      # @return [Enumerable<Spree::Image>] all images in the gallery
+      def images
+        @variant.images
+      end
+
+      # The "primary" (first) {Spree::Image} associated with the gallery
+      #
+      # @return [Spree::Image] primary image for the gallery
+      def primary_image
+        @variant.images.first || Spree::Image.new
+      end
+
+      # The "best" {Spree::Image} associated with the gallery
+      # This will return the first associated product image if there are no
+      # variant images
+      #
+      # @return [Spree::Image] best image for the variant gallery
+      def best_image
+        @variant.images.first || product_images.first || Spree::Image.new
+      end
```

Comment:
Why not remove these image collector methods from here and move into the views. This might be more duplication, but is also more flexible. Although I think the naming is difficult and misleading. Let’s not make these decisions here, but in the views. They are easier to override in a per use case basis.